design/pen.behavior.md

```diff
@@ -0,0 +1,79 @@
+# Pen Behavior
+
+The pen behavior is a method of recording the path by which a user moves a grabbed object that they activate so that various scenearios 
+can be carried out that wish to capture the virtual drawing of a user.  This behavior will carry with it client side visuals
+to ensure low-latency feedback, but will supply the complete curve data needed by the MRE app to acto on once the drawing is done.
+The `PenBehavior` will be an extension of the `ToolBehavior`.
+
+## Architecture
+
+There will be a new `PenBehavior` that extends the `ToolBehavior` that will have the ability to generate a wet ink effect along the use 
+path on the client and provide the use path to the MRE app for generating a spline curve asset that could have a mesh to applied to it
+before sending it to all the clients to render. and a mesh to apply to this spline curve.
+
+### Behavior
+``` ts
+/**
+ * Pen behavior class containing the target behavior actions.
+ */
+export class PenBehavior extends ToolBehavior {
+	public drawpointLocalOffset: ScaledTransform;
```

Comment:
Instead of a transform here, maybe it would be a child actor ID, whose transform we use? (or 0,0,0 if not set). I feel like having to add a local coordinates feels more imprecise. I can't think of other places where we input offsets anywhere in code (other than the actor's own transform, of course), and that's not a pattern I think we should create.

Reply:
We need some kind of offset.  Whether an actor or an offset.  The flexibility of a transform offset is that you do not need to create an additional actor just to be able to use the pen tool.  That seems like unnecessary overhead to me in trying to make an actor behave like a pen.  The local offset is only used if set in this case.  If you are supplying the model, you can make your offset very precise.